src/simulation/simulation.jl

```diff
@@ -196,49 +205,99 @@ function _check_folder(sim::Simulation)
+            set_status!(model, BuildStatus.FAILED)
+            rethrow(e)
+        end
+    end
+    return
+end
+
+function _build_emulation_model(sim::Simulation)
+    model = get_emulation_model(get_models(sim))
+
+    if model === nothing
+        return
+    end
+
+    try
+        @show get_execution_count(model)
+        TimerOutputs.@timeit BUILD_PROBLEMS_TIMER "Problem $(get_name(model))" begin
+            build_impl!(model)
         end
-        _populate_caches!(sim, name)
         sim.internal.date_ref[model_number] = initial_time
+        set_status!(model, BuildStatus.BUILT)
+    catch e
+        set_status!(model, BuildStatus.FAILED)
+        rethrow(e)
     end
     return
 end
 
+function _initialize_simulation_state(sim::Simulation) end
+
 function _build!(sim::Simulation, serialize::Bool)
     set_simulation_build_status!(sim, BuildStatus.IN_PROGRESS)
     problem_initial_times = _get_simulation_initial_times!(sim)
     sequence = get_sequence(sim)
-    for (ix, model) in enumerate(get_models(sim))
-        name = get_name(model)
-        problem_interval = get_interval(sequence, name)
+    step_resolution = get_step_resolution(sequence)
+    simulation_models = get_models(sim)
+    for (ix, model) in enumerate(get_decision_models(simulation_models))
+        problem_interval = get_interval(sequence, model)
         # Note to devs: Here we are setting the number of operations problem executions we
-        # will see for every step of the simulation
+        # will see for every step of the simulation. The step of the simulation is determined
+        # by the first decision problem interval
         if ix == 1
             set_executions!(model, 1)
         else
-            step_resolution = get_step_resolution(sequence)
+            if step_resolution % problem_interval != Dates.Millisecond(0)
+                error(
+                    "The $(get_name(model)) problem interval is not an integer fraction of the simulation step",
+                )
+            end
             set_executions!(model, Int(step_resolution / problem_interval))
         end
     end
+
+    em = get_emulation_model(simulation_models)
+    if em !== nothing
+        system = get_system(em)
+        em_resolution = PSY.get_time_series_resolution(system)
+        set_executions!(em, Int(step_resolution / em_resolution))
+    end
+
     TimerOutputs.@timeit BUILD_PROBLEMS_TIMER "Check Steps" begin
         _check_steps(sim, problem_initial_times)
     end
+
     TimerOutputs.@timeit BUILD_PROBLEMS_TIMER "Build Problems" begin
-        _build_problems!(sim, serialize)
-        # Make EmulationModel
-        # Make SimulationState here
+        _build_decision_models!(sim)
+        _build_emulation_model(sim)
+    end
+
+    TimerOutputs.@timeit BUILD_PROBLEMS_TIMER "Initialize Simulation State" begin
+        _initialize_simulation_state(sim)
     end
+
+
+    # _initialize_problem_storage!(sim, cache_size_mib, min_cache_flush_size_mib)
```

Comment:
@daniel-thom The storage should be instantiated here. The code is pretty similar to what we will be doing for the `simulation_state`

Reply:
We discussed this over the phone and decided to just instantiate all the store params and not the store itself here.

Reply:
Should these lines be deleted?

Reply:
Yeah, this will be substituted with the correct function call.